django_cte/cte.py

```diff
@@ -125,7 +125,20 @@ class CTEManager(Manager):
     """Manager for models that perform CTE queries"""
 
     def get_queryset(self):
-        return CTEQuerySet(self.model, using=self._db)
+        qs = super(CTEManager, self).get_queryset()
+        if isinstance(qs, CTEQuerySet):
+            return qs
+        else:
+            assert type(qs) == QuerySet, \
+                "models with CTE support need to use a CTEQuerySet"
+            return CTEQuerySet(self.model, using=self._db)
+
+    @classmethod
+    def from_queryset(cls, queryset_class, class_name=None):
+        assert issubclass(queryset_class, CTEQuerySet), \
+            "models with CTE support need to use a CTEQuerySet"
```

Comment:
Same here: `TypeError` rather than `AssertionError`

Reply:
I'm happy with this, I now raise TypeError.
Django seams to use both on model class creation which is where this will happen.